src/main/java/com/linkedin/kmf/services/KeyedConsumeService.java

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright 2016 LinkedIn Corp. Licensed under the Apache License, Version 2.0 (the "License"); you may not use this
+ * file except in compliance with the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.linkedin.kmf.services;
+
+import com.linkedin.kmf.common.DefaultTopicSchema;
+import com.linkedin.kmf.common.Utils;
+import com.linkedin.kmf.consumer.BaseConsumerRecord;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.avro.generic.GenericRecord;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class KeyedConsumeService extends ConsumeService implements Service {
+  private static final Logger LOG = LoggerFactory.getLogger(KeyedConsumeService.class);
+
+  public KeyedConsumeService(Map<String, Object> props, String name) throws Exception {
+    super(props, name);
+  }
+
+  private void consume() throws Exception {
+    Map<String, Long> nextIndexes = new HashMap<>();
```

Comment:
This method in the original consumer could reused by making this Map<Object, Long>  not elegant, but probably ok.  The other possibility would be type parameterize the consumer, but that seems like more work for little gain.